### PR TITLE
Remove corner mask in favor of rounded background corners

### PR DESCRIPTION
**Pull request by leolost2605**
By making the system background entirely black we can remove the maskcorners plugin since we already round the background corners.

The only difference is that now windows or anything that's in the corner doesn't get rounded so I guess that's a design question 🤷

### CODE REVIEW

**Review by danirabbit — APPROVED** [review]
So the only place you notice this afaict is like fullscreen apps. Which I'm fine with fullscreen apps figuring out their own corners. This looks good to me! I'm happy if @lenemter is happy :)

**Review by lenemter — CHANGES_REQUESTED** [review]
You also need to deprecate the gsettings keys the plugin used

**Comment by leolost2605** [comment]
@lenemter done!

**Comment by lenemter** [comment]
For some reason windows are still getting clipped at corners and corners stay black even if I change background color to the old gray one. So nothing has changed for me.

@leolost2605 Could you revert the change to the background color please? The color is still shown when switching workspaces and black looks bad here.

**Comment by leolost2605** [comment]
> For some reason windows are still getting clipped at corners and corners stay black even if I change background color to the old gray one. So nothing has changed for me.

I think that's because the plugin isn't actually uninstalled when you install the branch so it's still there. I tested it by disabling the plugin 🤷

> @leolost2605 Could you revert the change to the background color please? The color is still shown when switching workspaces and black looks bad here.

Ah right I almost thought I might have missed a use :) Will change it

**Comment by leolost2605** [comment]
@lenemter this should be fixed now :)

**Review by lenemter — CHANGES_REQUESTED** [review]
Nice! Using `background_color` is a good solution but now the black corners are visible during workspace switch animation. To fix this you can change their color to transparent.

**Comment by leolost2605** [comment]
You mean by workspace switch switch via touchpad swipe/super + left/right? 
Because they aren't black for me there 🤔

**Comment by lenemter** [comment]
@leolost2605 Yes

screenshot

It's more obvious if I change system bg to white:

screenshot white

**Comment by leolost2605** [comment]
@lenemter ok so I'm guessing this has something to do with whether you have workspaces on primary only. Btw how do I disable this cause I didn't find it anywhere? Is there a dconf key somewhere?

I pushed your mentioned fix, couldn't test it though. Also if you have a better name for the method just change it I just couldn't figure out a  good name :)

**Comment by lenemter** [comment]
@leolost2605 it's /org/gnome/mutter/workspaces-only-on-primary. I don't have second monitor, so I don't know how it would affect it

**Comment by lenemter** [comment]
@leolost2605 it crashes wm for me

```
(gdb) bt
#0  0x000055e194671399 in gala_background_container_black_background (self=0x55e1959408f0, black=0)
    at ../src/Background/BackgroundContainer.vala:52
#1  0x000055e194659a1a in gala_window_manager_gala_prepare_workspace_switch (self=0x55e19593b250, from=0, to=1, direction=META_MOTION_RIGHT)
    at ../src/WindowManager.vala:1941
#2  0x000055e19465c64a in gala_window_manager_gala_real_switch_workspace (base=0x55e19593b250, from=0, to=1, direction=META_MOTION_RIGHT)
    at ../src/WindowManager.vala:2121
#3  0x00007fc62b2f35d9 in meta_workspace_activate_with_focus (workspace=0x55e195596cd0, focus_this=0x0, timestamp=0)
    at ../src/core/workspace.c:684
#4  0x000055e19464d5c8 in gala_window_manager_gala_real_switch_to_next_workspace
    (base=0x55e19593b250, direction=META_MOTION_RIGHT, timestamp=0) at ../src/WindowManager.vala:594
#5  0x00007fc62c452bf3 in gala_window_manager_switch_to_next_workspace (self=0x55e19593b250, direction=META_MOTION_RIGHT, timestamp=0)
    at ../lib/WindowManager.vala:183
```

**Comment by leolost2605** [comment]
@lenemter IIUC if workspaces_on_primary_only is enabled the workspace switch only uses the backgroundmanager of the primary/only monitor. But if it's disabled it uses the whole backgroundcontainer (where the black background is set) no matter whether you have one or multiple monitors.

Also the crash should be fixed now

File: src/Background/BackgroundContainer.vala [hunk 1]
```diff
@@ -47,6 +48,10 @@ namespace Gala {
             monitor_manager.monitors_changed.disconnect (update);
         }
 
+        public void black_background (bool black) {
```
Comment (lenemter) [inline src/Background/BackgroundContainer.vala:51]: Method names usually use verbs. Rename this to `set_black_background` please :)

**Review by lenemter — COMMENTED** [review]
Looks very good. One last comment

**Comment by leolost2605** [comment]
Done!

Btw is workspaces on all monitors something we might want to support in the future? Because currently it's entirely broken with multiple monitors?

**Review by lenemter — APPROVED** [review]
:rocket:

**Comment by lenemter** [comment]
@leolost2605 Personally, I'd like to add support for https://github.com/elementary/gala/issues/26 when/if mutter implements this, and don't find workspaces on all monitors useful. Maybe @danirabbit has an opinion on this?

**Pull request merged**